### PR TITLE
[local_proxy]: install pg_session_jwt extension on demand

### PR DESCRIPTION
**Pull request by conradludgate**
Follow up on #9344. We want to install the extension automatically. We didn't want to couple the extension into compute_ctl so instead local_proxy is the one to issue requests specific to the extension.

depends on #9344 and #9395

### CODE REVIEW

File: proxy/src/serverless/backend.rs
Comment (conradludgate) [inline proxy/src/serverless/backend.rs:276]:
```suggestion
```

File: proxy/src/serverless/backend.rs
Comment (conradludgate) [inline proxy/src/serverless/backend.rs:284]:
```suggestion
```

File: proxy/src/compute_ctl/mod.rs
Comment (conradludgate) [inline proxy/src/compute_ctl/mod.rs:60]: blocked until #9344 is merged

File: proxy/src/compute_ctl/mod.rs
Comment (conradludgate) [inline proxy/src/compute_ctl/mod.rs:70]: blocked until #9395 is merged

File: proxy/src/compute_ctl/mod.rs
Comment (cloneable) [inline proxy/src/compute_ctl/mod.rs:51]: typo

File: proxy/src/serverless/backend.rs
Comment (cloneable) [inline proxy/src/serverless/backend.rs:264]: Still needed?

**Review by cloneable — APPROVED** [review]

File: proxy/src/compute_ctl/mod.rs
Comment (awarus) [inline proxy/src/compute_ctl/mod.rs:99]: Typo?

**Review by awarus — APPROVED** [review]

**Pull request merged**
